src/bitcoind/interface.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{bitcoind::BitcoindError, config::BitcoindConfig};
+
+use std::fs;
+
+use jsonrpc::{client::Client, simple_rtt::Tripper};
+
+pub struct BitcoinD {
+    // FIXME: do we need to persist the config here ?
+    client: Client<Tripper>,
+}
+
+impl BitcoinD {
+    pub fn new(config: &BitcoindConfig) -> Result<BitcoinD, BitcoindError> {
+        let cookie_string = fs::read_to_string(&config.cookie_path)
+            .map_err(|e| BitcoindError(format!("Reading cookie file: {}", e.to_string())))?;
+        // The cookie file content is "__cookie__:pass"
+        let mut cookie_slices = cookie_string.split(":");
+        let (user, pass) = (
+            cookie_slices.next().map(|s| s.to_string()),
+            cookie_slices.next().map(|s| s.to_string()),
+        );
+        let client = Client::new(format!("{}", config.addr), user, pass);
+
+        Ok(BitcoinD { client })
+    }
+
+    fn make_request<'a, 'b>(
+        &self,
+        method: &'a str,
+        params: &'b [serde_json::Value],
+    ) -> Result<serde_json::Value, BitcoindError> {
```

Comment:
I think you can change signature for:
```rust
use serde::de::DeserializeOwned;
use serde::{Deserialize, Serialize};
use serde_json;

    fn make _equest<P, T>(&self, method: &'a str, payload: P) -> Result<T, Error>
    where
        P: Serialize,
        T: DeserializeOwned,
    {
```

It will serialize the struct param and deserialize in the resource struct all using type inference

example:
https://github.com/edouardparis/opennode-rs/blob/master/opennode-client/src/client.rs#L88 
https://github.com/edouardparis/opennode-rs/blob/master/opennode-client/src/lib.rs#L51

Reply:
I can, not sure i need though ? It's like `do_rpc` in the crate i use